hyperspy/misc/array_tools.py

```diff
@@ -96,11 +97,106 @@ def rebin(a, new_shape):
     new_shape = tuple(int(ns) for ns in new_shape)
     factor = np.asarray(shape) // np.asarray(new_shape)
     evList = ['a.reshape('] + \
-             ['new_shape[%d],factor[%d],' % (i, i) for i in range(lenShape)] + \
+             ['new_shape[%d],factor[%d],' % (i, i) for i in range(lenShape)] +\
              [')'] + ['.sum(%d)' % (i + 1) for i in range(lenShape)]
     return eval(''.join(evList))
 
 
+def _linear_bin(s, scale,
```

Comment:
I haven't tried the code on anything larger than 600x600x2048 which seems to work in 2 seconds according to timeit:
import timeit
timeit.timeit(s.linear_bin(scale))
Out[22]:
2.048694275028538
But I guess that might depending on how much you're binning by. I agree faster would be better as I'm sure people would end up trying larger datasets than this. I would know where to start with optimising though. I'll try asking around the Jülich guys today. I would prefer to convert into C/Cython than vectorise if possible as I'm not sure I understand the vectorisation well enough to trust it's doing the exact correct thing. Currently the function is slow but 100% correct.